### PR TITLE
Cancel Boarding Modal change

### LINKED ISSUE #291: Cancel Boarding Modal change
When a user cancels a boarding they currently have a choice of 'Keep Editing' or 'Cancel Boarding'. Please add a third choice (in the middle) called 'Save and Finish Later' which will mark the data as a draft. Design is at https://mongodb.invisionapp.com/share/M6YN2Y77B8E#/screens/430515916_Modal-Discard_Boarding and the Realm flag for drafts was added in https://github.com/WildAid/o-fish-realm/issues/128

### PR DESCRIPTION
**Pull request by akaktus**
[-] Added ActionSheetItem.
[-] Implemented Cancel Boarding Model
[-] Implemented Save Boarding Modal
Fixes #291, #292, #293, #297

### CODE REVIEW

**Comment by o-fish-wildaid** [comment]
[Review PRs](https://trello.com/c/LIywX167/325-review-prs)

File: o-fish-ios/Views/Components/ActionSheetItem.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+extension View {
+
+    func showingActionSheet(actionSheetItem: Binding<ActionSheetItem?>) -> some View {
+        actionSheet(item: actionSheetItem) { actionSheetItem in
+            return ActionSheet(title: Text(actionSheetItem.title), message: Text(actionSheetItem.message), buttons: [
+                actionSheetItem.firstButton,
+                actionSheetItem.secondButton,
+                actionSheetItem.thirdButton
+            ])
```
[the remark this answers is not in the document]
Comment (am-MongoDB) [reply o-fish-ios/Views/Components/ActionSheetItem.swift:27]: When cancelling a report with missing data, the design asks for that to be included in the modal:

image

File: o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift [hunk 2]
```diff
@@ -16,6 +16,7 @@ struct ReportNavigationRootView: View {
     @State private var showingAlertItem: AlertItem?
     @State private var notFilledScreens: [String] = TopTabBarItems.allCases.map { $0.rawValue }
     @State private var prefilledCrewAvailable: Bool
+    @State private var showingActionSheetItem: ActionSheetItem?
```
Comment (am-MongoDB) [inline o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift:19]: The `Submit` button should be `Save` 
image

**Review by am-MongoDB — CHANGES_REQUESTED** [review]

File: o-fish-ios/Views/Components/ActionSheetItem.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (akaktus) [reply o-fish-ios/Views/Components/ActionSheetItem.swift:27]: According design, it should happen when we submit the report. When cancel you should see https://mongodb.invisionapp.com/share/M6YN2Y77B8E#/screens/430515916_Modal-Discard_Boarding

[the remark this answers is not in the document]
Comment (akaktus) [reply o-fish-ios/Views/Components/ActionSheetItem.swift:27]: > The design has the first option in bold. The design also uses a different shade of blue. Can those be achieved using `actionSheet`?
> image

I changed color of the text, but I can't make first option in bold using "actionSheet"

File: o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift [hunk 3]
```diff
@@ -84,32 +91,51 @@ struct ReportNavigationRootView: View {
...
         report.discard()
-        rootIsActive.toggle()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            self.showCanceledAlert(isDraft: isDraft)
+        }
     }
 
-    private func saveAlertClicked() {
+    private func saveAlertClicked(isDraft: Bool) {
+        let temporaryDraft = report.draft
```
[the remark this answers is not in the document]
Comment (akaktus) [reply o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift:120]: fixed

File: o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift [hunk 4]
```diff
@@ -84,32 +91,51 @@ struct ReportNavigationRootView: View {
...
         }
+        
         report.save()
         DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-            self.showSubmittedAlert()
+            if temporaryDraft {
+                self.showSubmittedAlert(isDraft: temporaryDraft)
+            } else {
+                self.showSubmittedAlert(isDraft: isDraft)
+            }
```
[the remark this answers is not in the document]
Comment (akaktus) [reply o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift:135]: Changed logic.

File: o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift (hunk 2 above)
Comment (akaktus) [reply o-fish-ios/Views/ReportFlow/ReportNavigationRootView.swift:19]: Fixed

**Review by am-MongoDB — APPROVED** [review]

**Pull request merged**